sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/logging/PipelineEvent.scala

```diff
@@ -31,12 +33,12 @@ import org.apache.spark.sql.pipelines.graph.QueryOrigin
  */
 case class PipelineEvent(
     id: String,
-    timestamp: String,
+    timestamp: Timestamp,
     origin: PipelineEventOrigin,
     level: EventLevel,
     message: String,
     details: EventDetails,
-    error: Option[ErrorDetail]
+    error: Option[Throwable]
```

Comment:
Changing this to a Throwable in favor of a custom class.

Reply:
Also removing a bunch of tests around the custom class and error serialization since that is no longer needed